apis/buzzer/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+//! Implementation started by : https://github.com/teodorobert
+//! Continued and modified by : https://github.com/SheepSeb
+#![no_std]
+
+use core::cell::Cell;
+use core::time::Duration;
+
+use libtock_platform::{share, DefaultConfig, ErrorCode, Subscribe, Syscalls};
+pub struct Buzzer<S: Syscalls>(S);
+
+impl<S: Syscalls> Buzzer<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// Initiate a tone
+    pub fn tone(freq: u32, duration: Duration) -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, BUZZER_ON, freq, duration.as_millis() as u32).to_result()
+    }
+
+    /// Register an events listener
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32,)>>,
```

Comment:
I would add a listener, as the listener should not return an u32, rather an Result<(), ErrorCode>

Reply:
I think we did this wrong here. We need to be able to register a listener that the user can actually use outside this crate, meaning that the listener needs to have a function that is called when the upcall is fired. 

Something similar to the [`ButtonListener`](https://github.com/tock/libtock-rs/blob/cff868b1fb2641f2d1b0f3edaae827cbcfdadef3/apis/buttons/src/lib.rs#L124) should work.